source/DistanceAndDirection/ProAppDistanceAndDirectionModule/ProAppDistanceAndDirectionModule/ViewModels/ProTabBaseViewModel.cs

```diff
@@ -474,17 +474,76 @@ private void OnClearGraphics()
         {
             if (MapView.Active == null)
                 return;
+            if (this is ProAppDistanceAndDirectionModule.ViewModels.ProCircleViewModel)
+            {
 
-            foreach (var item in GraphicsList)
+                foreach (var item in GraphicsList)
+                {
+                    if (item.GraphicType == GraphicTypes.Circle)
+                    {
+                        item.Disposable.Dispose();
+                    }
+                }
+                GraphicsList.RemoveAll(IsCircle);
+            }
+            else if (this is ProAppDistanceAndDirectionModule.ViewModels.ProLinesViewModel)
             {
-                item.Disposable.Dispose();
+                foreach (var item in GraphicsList)
+                {
+                    if (item.GraphicType == GraphicTypes.Line)
+                    {
+                        item.Disposable.Dispose();
+                    }
+                }
+                GraphicsList.RemoveAll(IsLine);
             }
-
-            GraphicsList.Clear();
+            else if (this is ProAppDistanceAndDirectionModule.ViewModels.ProEllipseViewModel)
+            {
+                foreach (var item in GraphicsList)
+                {
+                    if (item.GraphicType == GraphicTypes.Ellipse)
+                    {
+                        item.Disposable.Dispose();
+                    }
+                }
+                GraphicsList.RemoveAll(IsEllipse);
+            }
+            else if (this is ProAppDistanceAndDirectionModule.ViewModels.ProRangeViewModel)
+            {
+                foreach (var item in GraphicsList)
+                {
+                    if (item.GraphicType == GraphicTypes.RangeRing)
+                    {
+                        item.Disposable.Dispose();
+                        GraphicsList.Remove(item);
```

Comment:
Just confirming this line is intended/needed (since this is the only branch with this extra remove)